ts/state/ducks/search.ts

```diff
@@ -283,15 +281,13 @@ async function queryConversationsAndContacts(
     query
   );
 
-  const ourPrimaryDevice = isSecondaryDevice
-    ? await getPrimaryDeviceFor(ourNumber)
-    : ourNumber;
+  const ourPrimaryDevice = await MultiDeviceProtocol.getPrimaryDevice(
+    ourNumber
+  );
 
-  const resultPrimaryDevices: Array<string | null> = await Promise.all(
+  const resultPrimaryDevices = await Promise.all(
     searchResults.map(async conversation =>
-      conversation.id === ourPrimaryDevice
-        ? Promise.resolve(ourPrimaryDevice)
-        : getPrimaryDeviceFor(conversation.id)
+      MultiDeviceProtocol.getPrimaryDevice(conversation.id)
```

Comment:
I think we have a field on the conversation for the primaryDevice pubkey now. Should we use it there?

Reply:
hmm which field are you talking about?